dali/imgcodec/parsers/png.cc

```diff
@@ -12,19 +12,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "dali/core/byte_io.h"
 #include "dali/imgcodec/parsers/png.h"
 
 namespace dali {
 namespace imgcodec {
 
+// https://www.w3.org/TR/2003/REC-PNG-20031110
+
+enum ColorType : uint8_t {
+  PNG_COLOR_TYPE_GRAY       = 0,
+  PNG_COLOR_TYPE_RGB        = 2,
+  PNG_COLOR_TYPE_PALETTE    = 3,
+  PNG_COLOR_TYPE_GRAY_ALPHA = 4,
+  PNG_COLOR_TYPE_RGBA       = 6
+};
+
+struct IhdrChunk {
+  uint32_t width;
+  uint32_t height;
+  uint8_t color_type;
+  // Some fields were ommited.
+
+  int GetNumberOfChannels() {
```

Comment:
Perhaps we could use something like:
```suggestion
  int GetNumberOfChannels(bool include_alpha) {
```

Reply:
Should I use it later as follows?

```c++
info.shape = {
    ihdr.height,
    ihdr.width,
    ihdr.GetNumberOfChannels(true)
};
```

Reply:
I guess so...